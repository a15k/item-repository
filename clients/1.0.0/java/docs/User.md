
# User

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**id** | [**UUID**](UUID.md) |  | 
**role** | [**RoleEnum**](#RoleEnum) |  | 
**username** | **String** |  |  [optional]
**name** | **String** |  |  [optional]


<a name="RoleEnum"></a>
## Enum: RoleEnum
Name | Value
---- | -----
POWER_USER | &quot;power_user&quot;
STANDARD_USER | &quot;standard_user&quot;



