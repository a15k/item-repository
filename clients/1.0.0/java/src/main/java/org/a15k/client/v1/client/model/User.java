/*
 * Assessment Network API
 * > Stores content for the Assessment Network  API Version is specified in the URL path as `api/<version>/resource` Version will default to the latest if it's ommited  All endpoints require an API key to be passed in the request header. API keys can be obtained by members at www.a15k.org. 
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


package org.a15k.client.v1.client.model;

import java.util.Objects;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.io.IOException;
import java.util.UUID;

/**
 * A user who can login and use the AssessmentNetwork interactively
 */
@ApiModel(description = "A user who can login and use the AssessmentNetwork interactively")
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2018-11-14T10:14:33.057-06:00")
public class User {
  @SerializedName("id")
  private UUID id = null;

  /**
   * Gets or Sets role
   */
  @JsonAdapter(RoleEnum.Adapter.class)
  public enum RoleEnum {
    POWER_USER("power_user"),
    
    STANDARD_USER("standard_user");

    private String value;

    RoleEnum(String value) {
      this.value = value;
    }

    public String getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    public static RoleEnum fromValue(String text) {
      for (RoleEnum b : RoleEnum.values()) {
        if (String.valueOf(b.value).equals(text)) {
          return b;
        }
      }
      return null;
    }

    public static class Adapter extends TypeAdapter<RoleEnum> {
      @Override
      public void write(final JsonWriter jsonWriter, final RoleEnum enumeration) throws IOException {
        jsonWriter.value(enumeration.getValue());
      }

      @Override
      public RoleEnum read(final JsonReader jsonReader) throws IOException {
        String value = jsonReader.nextString();
        return RoleEnum.fromValue(String.valueOf(value));
      }
    }
  }

  @SerializedName("role")
  private RoleEnum role = null;

  @SerializedName("username")
  private String username = null;

  @SerializedName("name")
  private String name = null;

  public User id(UUID id) {
    this.id = id;
    return this;
  }

   /**
   * Get id
   * @return id
  **/
  @ApiModelProperty(required = true, value = "")
  public UUID getId() {
    return id;
  }

  public void setId(UUID id) {
    this.id = id;
  }

  public User role(RoleEnum role) {
    this.role = role;
    return this;
  }

   /**
   * Get role
   * @return role
  **/
  @ApiModelProperty(required = true, value = "")
  public RoleEnum getRole() {
    return role;
  }

  public void setRole(RoleEnum role) {
    this.role = role;
  }

  public User username(String username) {
    this.username = username;
    return this;
  }

   /**
   * Get username
   * @return username
  **/
  @ApiModelProperty(value = "")
  public String getUsername() {
    return username;
  }

  public void setUsername(String username) {
    this.username = username;
  }

  public User name(String name) {
    this.name = name;
    return this;
  }

   /**
   * Get name
   * @return name
  **/
  @ApiModelProperty(value = "")
  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    User user = (User) o;
    return Objects.equals(this.id, user.id) &&
        Objects.equals(this.role, user.role) &&
        Objects.equals(this.username, user.username) &&
        Objects.equals(this.name, user.name);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, role, username, name);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class User {\n");
    
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    role: ").append(toIndentedString(role)).append("\n");
    sb.append("    username: ").append(toIndentedString(username)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

