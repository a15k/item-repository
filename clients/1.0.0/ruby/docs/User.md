# A15kClient::User

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**id** | **String** |  | 
**role** | **String** |  | 
**username** | **String** |  | [optional] 
**name** | **String** |  | [optional] 


