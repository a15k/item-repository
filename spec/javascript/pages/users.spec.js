import renderer from 'react-test-renderer';
import Users from 'pages/users';
import Factory from '../factories';
import User from 'models/user';

jest.mock('models/user', () => ({
  id: 1,
}));
jest.mock('models/config', () => ({
  member_name: 'Illuminati',
}));
jest.mock('pages/users/delete-placeholder', () => {
  return () => <div className="delete-placeholder" />;
});

jest.mock('pages/users/disabled-power-user-toggle', () => {
  return () => <input type="checkbox" className="disabled-power-user" readOnly checked />;
});

describe(Users, () => {
  let props;

  const setSinglePowerUser = () => {
    const user = props.users.array[2];
    props.users.powerUsers = () => [ user ];
    props.users.map.replace({ [`${user.id}`]: user });
    return user;
  };

  beforeEach(() => {
    const users = Factory.usersCollection({ count: 3 });
    users.powerUsers = () => [ ];

    users.api.fetch = jest.fn(() => Promise.resolve({ data: [] }));
    props = {
      users,
    };
  });

  it('renders listing', () => {
    const users = renderer.create(<Users {...props} />);
    expect(users.toJSON()).toMatchSnapshot();
    users.unmount();
  });

  it('can delete a user', () => {
    const users = mount(<Users {...props} />);
    const user = props.users.array[1];
    const delBtn = users.find(`ListGroupItem[data-id="${user.id}"] SuretyGuard`);
    props.users.destroy = jest.fn();
    delBtn.props().onConfirm();
    expect(props.users.destroy).toHaveBeenCalledWith(user);
    users.unmount();
  });

  it('hides delete for last power user', () => {
    const user = setSinglePowerUser();
    const users = mount(<Users {...props} />);
    expect(users.find('ListGroupItem').length).toEqual(1);
    expect(users).toRender('DeleteButton Button');
    user.id = 1;
    users.update();
    expect(users).not.toRender('DeleteButton Button');
    users.unmount();
  });

  it('de-activates toggle for last power user', () => {
    const user = setSinglePowerUser();
    user.id = 1;
    const users = mount(<Users {...props} />);
    expect(users).toRender('PowerUserToggle input[readOnly]');
    users.unmount();
  });

  describe('invitations', () => {
    it('can invite users', () => {
      const users = mount(<Users {...props} />);
      users.instance().invite.value = 'newUserOne@test.com';
      props.users.invite = jest.fn(() => Promise.resolve());
      users.find('[icon="envelope"] .btn-secondary').simulate('click');
      expect(props.users.invite).toHaveBeenCalledWith('newUserOne@test.com');
      users.unmount();
    });

    it('displays message', async () => {
      const users = mount(<Users {...props} />);
      users.instance().invite.value = 'bob@test.com';
      props.users.invite = jest.fn(() => Promise.resolve());
      users.find('[icon="envelope"] .btn-secondary').simulate('click');
      expect(props.users.invite).toHaveBeenCalled();
      await testhelper.waitUntil(() => users.instance().inviteDeliveredTo)
      expect(users.find('Alert[color="info"]').text()).toContain('email has been sent to bob@test.com');
      users.unmount();
    });

    it('displays errors', () => {
      const users = mount(<Users {...props} />);
      users.instance().invite.value = 'not-an-email';
      props.users.invite = jest.fn(function() {
        this.errors = { base: 'unable to invite user' };
        return Promise.resolve();
      });
      users.find('[icon="envelope"] .btn-secondary').simulate('click');
      expect(props.users.invite).toHaveBeenCalled();
      expect(users.find('Alert[className="model-errors"]').text()).toEqual('unable to invite user');
      users.unmount();

    });


  });

});
