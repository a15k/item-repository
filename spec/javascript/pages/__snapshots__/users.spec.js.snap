// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders listing 1`] = `
<div
  className="access-users"
>
  <h1>
    Users
  </h1>
  <div
    className="d-flex justify-content-end"
  >
    PowerUser?
    <div
      className="sc-ifAKCX fNMcFr"
    >
      Delete
    </div>
  </div>
  <ul
    className="list-group"
  >
    <li
      className="d-flex align-items-center list-group-item"
      data-id="379d79cc-cd42-443c-9f2e-15087a5bc518"
    >
      <div
        className="sc-htpNat ccbDWo"
      >
        Mr. Imani Ankunding
      </div>
      <input
        checked={true}
        className="sc-bxivhb jBgGdK"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="sc-bdVaJa kaVmVf btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 icon"
          data-icon="trash-alt"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </li>
    <li
      className="d-flex align-items-center list-group-item"
      data-id="5b9e1f51-e6a7-468c-8579-6d96fc028b7b"
    >
      <div
        className="sc-htpNat ccbDWo"
      >
        Erling Walsh
      </div>
      <input
        checked={false}
        className="sc-bxivhb jBgGdK"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="sc-bdVaJa kaVmVf btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 icon"
          data-icon="trash-alt"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </li>
    <li
      className="d-flex align-items-center list-group-item"
      data-id="00f2b0ea-e9f8-4604-865c-89119b920095"
    >
      <div
        className="sc-htpNat ccbDWo"
      >
        Pattie Bernhard
      </div>
      <input
        checked={true}
        className="sc-bxivhb jBgGdK"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="sc-bdVaJa kaVmVf btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 icon"
          data-icon="trash-alt"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </li>
  </ul>
  <div
    className="invite-user input-group"
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        Invite email:
      </span>
    </div>
    <input
      className="form-control"
      name="email"
      type="email"
    />
    <div
      className="input-group-append"
    >
      <button
        className="sc-bdVaJa kaVmVf btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-envelope fa-w-16 icon"
          data-icon="envelope"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
